src/WebJobs.Script.WebHost/Diagnostics/EventGenerator.cs

```diff
@@ -9,114 +9,114 @@ namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
         }
 
-        public void LogFunctionDetailsEvent(string siteName, string functionName, string inputBindings, string outputBindings, string scriptType, bool isDisabled)
+        public void LogFunctionDetailsEvent(string siteName, string functionName, string hostVersion, string inputBindings, string outputBindings, string scriptType, bool isDisabled)
         {
-            Events.Instance.LogFunctionDetailsEvent(siteName, functionName, inputBindings, outputBindings, scriptType, isDisabled);
+            FunctionsEventSource.Instance.LogFunctionDetailsEvent(siteName, functionName, hostVersion, inputBindings, outputBindings, scriptType, isDisabled);
         }
 
-        public void LogFunctionExecutionEvent(string executionId, string siteName, int concurrency, string functionName, string invocationId, string executionStage, long executionTimeSpan, bool success)
+        public void LogFunctionExecutionEvent(string executionId, string siteName, int concurrency, string functionName, string hostVersion, string invocationId, string executionStage, long executionTimeSpan, bool success)
         {
-            Events.Instance.LogFunctionExecutionEvent(executionId, siteName, concurrency, functionName, invocationId, executionStage, (ulong)executionTimeSpan, success);
+            FunctionsEventSource.Instance.LogFunctionExecutionEvent(executionId, siteName, concurrency, functionName, hostVersion, invocationId, executionStage, (ulong)executionTimeSpan, success);
         }
 
         [EventSource(Guid = "08D0D743-5C24-43F9-9723-98277CEA5F9B")]
-        public sealed class Events : EventSource
+        public sealed class FunctionsEventSource : EventSource
         {
-            internal static readonly Events Instance = new Events();
+            internal static readonly FunctionsEventSource Instance = new FunctionsEventSource();
 
             [Event(57907, Level = EventLevel.Informational, Channel = EventChannel.Operational)]
-            public void LogFunctionExecutionAggregateEvent(string siteName, string functionName, ulong executionTimeInMs, ulong functionStartedCount, ulong functionCompletedCount, ulong functionFailedCount)
+            public void LogFunctionExecutionAggregateEvent(string siteName, string functionName, string hostVersion, ulong executionTimeInMs, ulong functionStartedCount, ulong functionCompletedCount, ulong functionFailedCount)
             {
-                if (IsEnabled())
-                {
-                    WriteEvent(57907, siteName, functionName, executionTimeInMs, functionStartedCount, functionCompletedCount, functionFailedCount);
-                }
+               if (IsEnabled())
+               {
+                   WriteEvent(57907, siteName, functionName, hostVersion, executionTimeInMs, functionStartedCount, functionCompletedCount, functionFailedCount);
+               }
```

Comment:
Yes. As ScriptHost.Version is available throughout the lifecycle of the App, this can simplified. Will look into it.